### PR TITLE
fix(Vulnerability card): Fix race condition leading to invalid affecting filter

### PR DESCRIPTION
**Pull request by leSamo**
https://issues.redhat.com/browse/RHINENG-12218

### CODE REVIEW

**Comment by mkholjuraev** [comment]
@leSamo can you provide more info on how to test the PR?

File: src/SmartComponents/Vulnerability/VulnerabilityCard.js [hunk 1]
```diff
@@ -162,7 +162,7 @@ const VulnerabilityCard = () => {
         }]);
 
         return [chartData, rows];
-    }, [vulnerabilitiesFetchStatus, vulnerabilities]);
+    }, [vulnerabilitiesFetchStatus, vulnerabilities, affectingFilterValue]);
```
Comment (bastilian) [inline src/SmartComponents/Vulnerability/VulnerabilityCard.js:165]: Hm. Why is this needed as a dependency if it is not used in the memo?

**Comment by leSamo** [comment]
@mkholjuraev this commit fixes the issue mentioned by Max in the Jira ticket comment. Before this commit the link sometimes has the incorrect filter applied (`affecting=true` when edge systems are present, which caused error in Vulnerability when the link is clicked). To test this I would say it's sufficient to refresh the Dashboard a few times on an account with edge systems and check if all of the links in Vulnerability card contain `affecting=rpmdnf,edge`

**Review by mkholjuraev — APPROVED** [review]
LGTM!
